Makefile "gen" target does not work with parallelization

Works fine if you just do `make gen` but if I try to get some parallelization going it always falls over.

```bash
coder@dev coder (main) make -j -B gen
codersdk/rbacresources_gen.go:1:1: expected 'package', found 'EOF'
codersdk/rbacresources_gen.go:1:1: expected 'package', found 'EOF'
make: *** [Makefile:553: coderd/database/dump.sql] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:617: examples/examples.gen.json] Error 1
make: *** Deleting file 'examples/examples.gen.json'
../../../../rbac/object_gen.go:1:1: expected 'package', found 'EOF'
+ pnpm install
Lockfile is up to date, resolution step is skipped
Already up to date
Done in 655ms
docs/admin/prometheus.md 114ms
```

cc @Emyrk you may know something about this

--- a/Makefile
+++ b/Makefile
@@ -629,7 +629,10 @@ coderd/rbac/object_gen.go: scripts/rbacgen/rbacobject.gotmpl scripts/rbacgen/mai
 	go run scripts/rbacgen/main.go rbac > coderd/rbac/object_gen.go
 
 codersdk/rbacresources_gen.go: scripts/rbacgen/codersdk.gotmpl scripts/rbacgen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
-	go run scripts/rbacgen/main.go codersdk > codersdk/rbacresources_gen.go
+	# Do no overwrite codersdk/rbacresources_gen.go directly, as it would make the file empty, breaking
+ 	# the `codersdk` package and any parallel build targets.
+	go run scripts/rbacgen/main.go codersdk > /tmp/rbacresources_gen.go
+	mv /tmp/rbacresources_gen.go codersdk/rbacresources_gen.go
 
 site/src/api/rbacresourcesGenerated.ts: scripts/rbacgen/codersdk.gotmpl scripts/rbacgen/main.go coderd/rbac/object.go coderd/rbac/policy/policy.go
 	go run scripts/rbacgen/main.go typescript > "$@"

